### PR TITLE
feat(nodeadm): configure container runtime based on binaries

### PR DESCRIPTION
**Pull request by ndbaker1**
**Issue #, if available:**

**Description of changes:**

This PR removes validation of instance type for loading the nvidia container runtime by relying on the presence of the `nvidia-container-runtime` binaries being present in the AMI. This is normally a safe assumption, since you should only be using NVIDIA-enabled EKS AMIs with NVIDIA-based EC2 instances.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Testing Done**

the only edge cases we really need to test is when the GPU AMI is used on a non-GPU instance type, because the assumptions is baked in that the `nvidia-container-runtime` exists, but we need to know if this introduces any conflicts.

*[See this guide for recommended testing for PRs.](../doc/CONTRIBUTING.md#testing-changes) Some tests may not apply. Completing tests and providing additional validation steps are not required, but it is recommended and may reduce review time and time to merge.*

### CODE REVIEW

**Comment by ndbaker1** [comment]
testing GPU AMI on non-GPU instance

/ci
+workflow:k8s_versions 1.32
+build enable_accelerator=nvidia enable_efa=true
+test --instance-types=m5.4xlarge

**Comment by ndbaker1** [comment]
test GPU types still work

/ci
+workflow:k8s_versions 1.32
+build enable_accelerator=nvidia enable_efa=true
+test --instance-types=g4dn.16xlarge

**Comment by ndbaker1** [comment]
the capacity didn't launch

/ci
+workflow:k8s_versions 1.32
+build enable_accelerator=nvidia enable_efa=true
+test --instance-types=g4dn.xlarge,g4dn.2xlarge,g4dn.4xlarge,g4dn.8xlarge,g4dn.16xlarge

File: nodeadm/internal/containerd/runtime_config_nvidia.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	nvidiaRuntimeBinaryPath = "/usr/bin/nvidia-container-runtime"
+)
+
+func NewNvidiaRuntimeConfigMixin() *nvidiaRuntimeConfigMixin {
+	runtimeOptionMixin := nvidiaRuntimeConfigMixin{
+		runtimeBinaryName: nvidiaRuntimeBinaryPath,
+	}
+	if binPath, ok := os.LookupEnv("NODEADM_NVIDIA_CONTAINER_RUNTIME_PATH"); ok {
+		runtimeOptionMixin.runtimeBinaryName = binPath
+	}
```
Comment (cartermckinnon) [inline nodeadm/internal/containerd/runtime_config_nvidia.go:21]: could you override `runtimeBinaryName` in your unit tests and avoid adding this fallback to prod?

File: nodeadm/internal/containerd/runtime_config.go [hunk 2]
```diff
@@ -1,65 +1,35 @@
...
+// getRuntimeOptions adds the needed OCI hook options to containerd config.toml
+// based on the instance family and available runtime binaries
+func getRuntimeOptions(cfg *api.NodeConfig) runtimeConfig {
+	options := runtimeConfig{
+		RuntimeName:       defaultRuntimeName,
+		RuntimeBinaryPath: defaultRuntimeBinaryPath,
+	}
+	for _, mixin := range []runtimeConfigMixin{
+		NewNvidiaRuntimeConfigMixin(),
+	} {
```
Comment (cartermckinnon) [inline nodeadm/internal/containerd/runtime_config.go:29]: nit: would declare this []mixin at package scope

Comment (ndbaker1) [reply nodeadm/internal/containerd/runtime_config.go]: updated

**Comment by ndbaker1** [comment]
/ci
+workflow:k8s_versions 1.32
+build enable_accelerator=nvidia enable_efa=true
+test --instance-types=g5.xlarge

**Comment by ndbaker1** [comment]
increased ec2 limits first and retrying

/ci
+workflow:k8s_versions 1.32
+build enable_accelerator=nvidia enable_efa=true
+test --instance-types=g4dn.xlarge,g4dn.2xlarge,g4dn.4xlarge,g4dn.8xlarge,g4dn.16xlarge

**Review by mselim00 — APPROVED** [review]
lgtm, seems like a great way to get rid of a hardcoded list using a cheap call

**Comment by wwvela** [comment]
saw the test has passed in latest ci https://github.com/awslabs/amazon-eks-ami/actions/runs/14764397846/job/41452518330. Should be good to merge.
